pandas/core/arrays/masked.py

```diff
@@ -155,8 +148,9 @@ def _from_sequence(cls, scalars, *, dtype=None, copy: bool = False) -> Self:
     @classmethod
     @doc(ExtensionArray._empty)
     def _empty(cls, shape: Shape, dtype: ExtensionDtype) -> Self:
-        values = np.empty(shape, dtype=dtype.type)
-        values.fill(cls._internal_fill_value)
+        dtype = cast(BaseMaskedDtype, dtype)
```

Comment:
The cast suspicious - why wouldn't we just change the type of `dtype` to `BaseMaskedType` if the `_internal_fill_value` attribute was required?

Reply:
IIRC mypy complains if we do that, but in practice that is correct that we should only ever get BaseMaskedDtype here